tomb

```diff
@@ -2935,6 +2935,160 @@ resize_tomb() {
+		_warning "or issue the command 'tomb suspend all' to suspend them all."
+		_failure "Operation aborted." }
+
+	for t in ${mounted_tombs}; do
+		mapper=`basename ${t[(ws:;:)1]}`
+
+		# strip square parens from tombname
+		tombname=${t[(ws:;:)5]}
+		tombmount="${t[(ws:;:)2]}"
+		tombfs=${t[(ws:;:)3]}
+		tombfsopts=${t[(ws:;:)4]}
+		tombloop=${mapper[(ws:.:)4]}
+
+		_verbose "Name: ::1 tomb name::" $tombname
+		_verbose "Mount: ::1 mount point::" "$tombmount"
+		_verbose "Loop: ::1 mount loop::" $tombloop
+		_verbose "Mapper: ::1 mapper::" $mapper
+
+		[[ -e "$mapper" ]] && {
+			_warning "Tomb not found: ::1 tomb file::" $1
+			_warning "Please specify an existing tomb."
+			return 0 }
+
+		#option_is_set -n || {
+		#	exec_safe_func_hooks \
+		#		close "$tombmount" "$tombname" "$tombloop" "$mapper"
+		#	exec_hook_res=$?
+		#	[[ $exec_hook_res = 0 ]] || {
+		#		_warning "close exec-hook returns a non-zero error code: ::1 error::" $exec_hook_res
+		#		_failure "Operation aborted"
+		#	}
+		#}
+
+		# kill all programs using the tomb
+		#	_message "Closing programs using files in tomb ::1 tombname:: mounted on ::2 tombmount::" \
+		#		 ${tombname} "${tombmount}"
+		#	_kill_processes "$tombname" "$tombmount"
+		#	[[ $? -ne 0 ]] && {
+		#		_failure "Still active processes for ::1 tombname ::, tomb not completly hidden." "$tombname"
+		#	}
+		#	_message "Suspending tomb ::1 tomb name:: mounted on ::2 mount point::" \
+		#		 $tombname "$tombmount"
+
+		# check if there are bind mounted dirs and close them first
+		# Can be due to bind-hooks or outside --bind mounts
+		#bind_tombs=(`list_tomb_binds "$mapper"`)
+		#for b in ${bind_tombs}; do
+		#	bind_mapper="${b[(ws:;:)1]}"
+		#	bind_mount="${b[(ws:;:)2]}"
+		#	_message "Closing tomb bind hook: ::1 hook::" "$bind_mount"
+		#	_sudo umount "$(echo "$bind_mount")" ||
+		#		_failure "Tomb bind hook ::1 hook:: is busy, cannot suspend tomb." "$bind_mount"
+		#done
+
+		# suspend the mapper
+		_sudo cryptsetup luksSuspend $mapper ||
+			_failure "Error occurred in cryptsetup luksSuspend ::1 mapper::" $mapper
+
+		# Avoid top level listing via a tmpfs mount
+		_sudo mount -t tmpfs -o size=10M,mode=1777 temporary $tombmount ||
+			_warning "Couldn't hide top level listing while suspending ::1 tombname::" "$tombname"
+
+		_success "Tomb ::1 tomb name:: suspended: your bones rest in peace for the time being." $tombname
+
+	done # loop across mounted tombs
+
+	return 0
+}
+
+# $1 = tombfile
+# ToDo
+# * what should be done with bind-hooks?
+resume_tomb() {
+	local mapper tombname tombmount
+
+	[[ -n "$1" ]] || _failure "No tomb name specified for resuming."
+	tombname=$1
+	mounted_tombs=(`list_tomb_mounts $tombname`)
+	for t in ${mounted_tombs}; do
+		mapper=`basename ${t[(ws:;:)1]}`
+		tombmount="${t[(ws:;:)2]}"
+		_verbose "Mapper: ::1 mapper::" $mapper
+	done
+
+	_message "Commanded to revive tomb ::1 tomb name::" $tombname
+	# Currently missing check if $tombname is valid or could be found
```

Comment:
Can this work?
```
if [[ ${#mounted_tombs[@]} -eq 0 ]]; then
	_failure "No mounted tombs found for: $tombname"
fi
```

Reply:
As a general check, yes. But it would still need to check on available/open mounts and their names in detail.